tools/bumper/component_commands.go

```diff
@@ -216,14 +215,21 @@ func (componentOps *gitComponent) getLatestTaggedFromBranch(repo, owner, branch,
 			return "", "", errors.Wrap(err, "Failed to get commit-sha from tag name")
 		}
 
-		for _, commit := range branchCommits {
-			if commit.GetSHA() == commitShaOfTag {
-				return tagName, commit.GetSHA(), nil
+		for pageIdx := 0; pageIdx < maxPagePaginate; pageIdx++ {
```

Comment:
the fact it is in a loop means we reget it for each tag
we can optimize it, question is if we want to

one way is that if it didnt find in the first page, fetch the pages but keep them in memory

Reply:
no, we stop as soon as we find a match. 
Most times the first tag searched is the one that will hit.

Reply:
i know, but if it isnt found for the first ttag we refetch them every loop this what i mean
anyhow we can live with that